internal/provider/resource/secret_resource.go

```diff
@@ -29,15 +29,21 @@ type secretResource struct {
 	client *infisical.Client
 }
 
+type SecretReminder struct {
+	Note       types.String `tfsdk:"note"`
+	RepeatDays types.Int64  `tfsdk:"repeat_days"`
+}
+
 // secretResourceSourceModel describes the data source data model.
 type secretResourceModel struct {
-	FolderPath  types.String `tfsdk:"folder_path"`
-	EnvSlug     types.String `tfsdk:"env_slug"`
-	Name        types.String `tfsdk:"name"`
-	Value       types.String `tfsdk:"value"`
-	WorkspaceId types.String `tfsdk:"workspace_id"`
-	LastUpdated types.String `tfsdk:"last_updated"`
-	Tags        types.List   `tfsdk:"tag_ids"`
+	FolderPath     types.String    `tfsdk:"folder_path"`
+	EnvSlug        types.String    `tfsdk:"env_slug"`
+	Name           types.String    `tfsdk:"name"`
+	SecretReminder *SecretReminder `tfsdk:"secret_reminder"`
```

Comment:
You don't need the `*SecretReminder`. You can just refer to the type like so `SecretReminder SecretReminder `tfsdk:"secret_reminder"`

Reply:
When using `SecretReminder SecretReminder`, I get this when a reminder isn't present in a secret
```
Error: Value Conversion Error
│ 
│   with infisical_secret.mongo_secret,
│   on resource.tf line 32, in resource "infisical_secret" "mongo_secret":
│   32: resource "infisical_secret" "mongo_secret" {
│ 
│ An unexpected error was encountered trying to build a value. This is always an error in the provider. Please report the following to
│ the provider developer:
│ 
│ Received null value, however the target type cannot handle null values. Use the corresponding `types` package type, a pointer type or a
│ custom type that handles null values.
│ 
│ Path: secret_reminder
│ Target Type: resource.SecretReminder
│ Suggested `types` Type: basetypes.ObjectValue
│ Suggested Pointer Type: *resource.SecretReminder

Reply:
Because of our API design, i think it makes sense to check for this as null. Resolving